.github/workflows/upload-and-deploy.yaml

```diff
@@ -68,6 +68,35 @@ jobs:
       - name: Deploy sceptre stacks
         run: pipenv run sceptre --var "namespace=${{ env.NAMESPACE }}" launch develop --yes
 
+  integration-test:
+    name: Deploys test events from the workflow
+    runs-on: ubuntu-latest
+    needs: [pre-commit, upload-files, sceptre-deploy-branch]
+    env:
+      NAMESPACE: $GITHUB_REF_NAME
+      LEMBDA_ENV_VARS_FILE: namespace_vars.json
+    if: github.ref_name != 'main' && github.ref_type != 'tag'
+    environment: develop
+    steps:
+
+      - name: Setup code, pipenv, aws
```

Comment:
These first two steps are duplicated on line 29 and 36. Let's move them to their own job and include it in depends.

Reply:
Will that work?  I feel like the environment of each gh-action step doesn't propogate.  Maybe I'm misunderstanding here.

Reply:
Tom and I were under the same impression here.  I do like your approach to cache the dependencies across runs.  I'll add the cache: pipenv to it.

Reply:
Thanks both, I filed a JIRA: https://sagebionetworks.jira.com/browse/IBCDPE-290